packages/backend/src/database/migrations/20250807212731_add_custom_roles.ts

```diff
@@ -0,0 +1,184 @@
+import { Knex } from 'knex';
+import { type DbScopeInsert } from '../entities/roles';
+
+const ScopesTableName = 'scopes';
+const RolesTableName = 'roles';
+const ScopedRolesTableName = 'scoped_roles';
+
+export async function up(knex: Knex): Promise<void> {
+    // Create roles table
+    await knex.schema.createTable(RolesTableName, (table) => {
+        table
+            .uuid('role_uuid')
+            .primary()
+            .defaultTo(knex.raw('uuid_generate_v4()'));
+        table.string('name', 100).notNullable();
+        table.text('description').nullable();
+        table
+            .uuid('organization_uuid')
+            .notNullable()
+            .references('organization_uuid')
+            .inTable('organizations')
+            .onDelete('CASCADE');
+        table
+            .uuid('created_by')
+            .references('user_uuid')
+            .inTable('users')
+            .onDelete('SET NULL');
+        table.timestamp('created_at').defaultTo(knex.fn.now());
+        table.timestamp('updated_at').defaultTo(knex.fn.now());
+
+        table.unique(['name', 'organization_uuid']);
+    });
```

Comment:
are we planning to do filtering by org_uuid ? for example, to get all roles per org, in that case. I'd suggest to add an index on `organization_uuid`

Reply:
Ah yep! I have an update coming.